src/main/java/com/google/finapp/Args.java

```diff
@@ -0,0 +1,31 @@
+package com.google.finapp;
+
+import com.beust.jcommander.Parameter;
```

Comment:
Hm, I have been looking into the interface of Commandline and it doesn't seem to offer any notable advantages. Is there a reason you had in mind to prefer the apache CLI? 

Here are a few things that stood out to me when I tried to convert this to apache CLI:

1. Parsing flags into required types is non-trivial. Example, if I want an int flag, I'd have to do something like:
```
// Specify an option which you use to build the commandline
OptionBuilder
    .withLongOpt("port-option")
    .withType(Number.class)
    .withArgName("port")
    .create()
// To parse the option
int port = ((Number)cmdLine.getParsedOptionValue("port-option")).intValue();
```
The use of "Number" type isn't obvious. Even otherwise, the casting and extraction of int from that doesn't seem very user friendly. 
JCommander on the other hand handles all of this for you at the framework level.

2. There's no built-in way to specify default values. We'd have to do this ourselves. Example:

```
int port = 8080; // initialize to some meaningful default value
if (cmdLine.hasOption("port")) {
    value = ((Number)cmdLine.getParsedOptionValue("port-option")).intValue();
}
```
Jcommander supports this at the framework level.

3. In general apache CLI seems to add more boilerplate code to achieve the same functionality.

Apache:
```
Options options = new Options();
options.addOption(
   Option.builder("p")
            .longOpt("port")
            .required(true)
            .type(Number.class)
            .hasArg()
            .build())

int port = 8080;
if (command.hasOption("port")) {
   port = ((Number) cmd.getParsedOptionValue("p")).intValue();
}
```

JCommander:
```
// Setting up options and using them
private static class Args {
    @Parameter(names = {"--port", "-p"})
    int port = 8080;
}

Args args = new Args();
JCommander.newBuilder().addObject(args).build().parse(argv);
int port = args.port;
```

Given all of the above, I personally would prefer using JCommander consistently throughout this project. We have a bug open to converge usages of JCommander and apache CLI (we use the latter in workload).

Reply:
I do not have a strong preference, I just saw org.apache.commons.cli.CommandLine used in other open source projects (e.g: PgAdapter). Let's leave as is, thanks for the rationale.

Some additional comments on how I saw this used:
I agree that both types and default values are not handled well in the framework level, but these have quite good handling in java. This is how I saw it used:
```
// Definition
options.addOption(OPTION_SERVER_PORT, "server-port", true,
        "This proxy's port number (Default 5432).");
```
```
// Parsing
int port = Integer.parseInt(
        commandLine.getOptionValue(
            OPTION_SERVER_PORT,
            DEFAULT_PORT));
```
which doesn't seem to add a lot of boilerplate, perhaps the errors are not as descriptive though.

I agree jcommander does more on the framework level, but I did find some errors quite confusing, esp. around arrity. In any case, there's no strong reason to switch. LGTM as is.

Reply:
Ah I see. Thanks for the explanation! I'll leave this as is, then.